cfme/tests/storage/test_volume.py

```diff
@@ -92,29 +134,62 @@ def test_storage_volume_crud(appliance, provider):
         casecomponent: Cloud
     """
     # create volume
-    volume_collection = appliance.collections.volumes
-    manager_name = '{} Cinder Manager'.format(provider.name)
-    volume = volume_collection.create(name=fauxfactory.gen_alpha(),
-                                      storage_manager=manager_name,
-                                      tenant=provider.data['provisioning']['cloud_tenant'],
-                                      size=STORAGE_SIZE,
-                                      provider=provider)
-    assert volume.exists
+    volume = create_volume(appliance, provider, from_manager, should_assert=True)
 
     # update volume
     old_name = volume.name
     new_name = fauxfactory.gen_alpha()
-    with update(volume):
-        volume.name = new_name
-
-    with update(volume):
-        volume.name = old_name
+    updates = {'volume_name': new_name, 'volume_size': STORAGE_SIZE + 1}
+    if provider.one_of(OpenStackProvider):
+        updates = {'volume_name': new_name}
+
+    if from_manager:
+        storage_manager = appliance.collections.block_managers.filter(
+            {"provider": provider}).all()[0]
+    else:
+        storage_manager = None
+    volume = volume.update(updates, storage_manager)
+    if provider.one_of(EC2Provider):
+        wait_for(lambda: volume.size == '{} GB'.format(updates.get('volume_size')), delay=15,
+                 timeout=900)
+
+    updates = {'volume_name': old_name}
+    volume = volume.update(updates, storage_manager)
 
     # delete volume
-    volume.delete(wait=True)
+    volume.delete(wait=True, storage_manager=storage_manager)
     assert not volume.exists
 
 
+@pytest.mark.uncollectif(lambda appliance, from_manager: appliance.version < '5.10' and
+                         from_manager == "from_manager")
+@pytest.mark.meta(blockers=[BZ(1684939, forced_streams=["5.9", "5.10", "upstream"],
+                               unblock=lambda provider: provider.one_of(EC2Provider))])
+@pytest.mark.tier(1)
+def test_storage_volume_attach_detach(appliance, provider, instance_fixture, from_manager):
+    volume = create_volume(appliance, provider, from_manager, az=instance_fixture.
+                           vm_default_args["environment"]["availability_zone"], should_assert=True)
+    storage_manager = appliance.collections.block_managers.filter(
```

Comment:
Optional:
@mmojzis what i think better way to handle this is `VolumeCollection` have a one property of `manager` like
https://github.com/ManageIQ/integration_tests/pull/8522/files#diff-91c1aeae2f85f62c8e46857e24985221R136

Reply:
That makes sense. I will add task for it to implement in the future.